src/ripple/overlay/impl/Manifest.cpp

```diff
@@ -275,6 +276,17 @@ ManifestDisposition
 ManifestCache::applyManifest (
     Manifest m, ValidatorList& unl, beast::Journal journal)
 {
+    /*
+        Move master public key from permanent trusted key list
+        to manifest cache.
+    */
+    if (unl.trusted(m.masterKey))
+    {
+        auto comment = unl.member (m.masterKey);
+        addTrustedKey (m.masterKey, comment.get());
+        unl.removePermanentKey (m.masterKey);
+    }
```

Comment:
Hmmm, I think we would `throw` on a restart.
It looks like the sequence would be:

1. Load [validators] to permanent list
2. Load manifests from db (add master keys to cache)
3. If/when we get another manifest for that master key, we would attempt to move it from permanent list to master cache, but it would already be on both.

I think this can be avoided by doing this same check in `ManifestCache::load`